submissions/api.py

```diff
@@ -373,6 +383,40 @@ def get_submissions(student_item_dict, limit=None):
     return SubmissionSerializer(submission_models, many=True).data
 
 
+def get_all_submissions(course_id, item_id, item_type, read_replica=True):
+    """
+    For the given item, get the most recent submission for every student who has submitted.
+
+    This may return a very large result set! It is implemented as a generator for efficiency.
+    It yields namedtuples with the following fields:
+        student_id
+        attempt_number
+        submitted_at
+        created_at
+        answer
+
+    Cannot fail unless there's a database error, but may return an empty list.
+    """
+    query = Submission.objects.filter(
+        student_item__course_id=course_id,
+        student_item__item_id=item_id,
+        student_item__item_type=item_type,
+    ).values_list(
+        'student_item__student_id', 'attempt_number', 'submitted_at', 'created_at', 'raw_answer'
+    ).order_by('student_item__student_id', '-created_at')
```

Comment:
Please make sure:

1. We have a test that asserts that only one query is happening here (so we're not doing n+1 fetches for the join)
2. The query plan for this looks sane for a large data set. If it's going to have to do a file sort or something like that, it might be best to just do the tracking of "this was the most recent entry for this user" in memory. You can still make sure that you only show only the most recent by doing an order_by on the `-id` for the Submission model, and only showing the first one that pops up. This wouldn't give us a nice ordering by student ID, but I think that's a worthwhile tradeoff if it makes the query a lot faster.